From d87e9fa711a64f0cc43a5f00bde84ca7cc3e61e7 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 13:11:48 -0400
Subject: [ethernet] i40evf: Use pci_enable_msix_range() instead of pci_enable_msix()

Message-id: <1409663591-32531-66-git-send-email-sassmann@redhat.com>
Patchwork-id: 90543
O-Subject: [RHEL7.1 PATCH 065/148] i40evf: Use pci_enable_msix_range() instead of pci_enable_msix()
Bugzilla: 1091127
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Alexander Gordeev <agordeev@redhat.com>

As result of deprecation of MSI-X/MSI enablement functions
pci_enable_msix() and pci_enable_msi_block() all drivers
using these two interfaces need to be updated to use the
new pci_enable_msi_range()  or pci_enable_msi_exact()
and pci_enable_msix_range() or pci_enable_msix_exact()
interfaces.

Cc: Jesse Brandeburg <jesse.brandeburg@intel.com>
Cc: linux-pci@vger.kernel.org
Signed-off-by: Alexander Gordeev <agordeev@redhat.com>
Tested-by: Sibai Li <sibai.li@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit fc2f2f5dfd4438bd785e773e46adf425b1e353f1)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40evf/i40evf_main.c b/drivers/net/ethernet/intel/i40evf/i40evf_main.c
index 13989ef..6f6bd3f 100644
--- a/drivers/net/ethernet/intel/i40evf/i40evf_main.c
+++ b/drivers/net/ethernet/intel/i40evf/i40evf_main.c
@@ -1032,30 +1032,21 @@ i40evf_acquire_msix_vectors(struct i40evf_adapter *adapter, int vectors)
   * Right now, we simply care about how many we'll get; we'll
   * set them up later while requesting irq's.
   */
- while (vectors >= vector_threshold) {
-  err = pci_enable_msix(adapter->pdev, adapter->msix_entries,
-          vectors);
-  if (!err) /* Success in acquiring all requested vectors. */
-   break;
-  else if (err < 0)
-   vectors = 0; /* Nasty failure, quit now */
-  else /* err == number of vectors we should try again with */
-   vectors = err;
- }
-
- if (vectors < vector_threshold) {
+ err = pci_enable_msix_range(adapter->pdev, adapter->msix_entries,
+        vector_threshold, vectors);
+ if (err < 0) {
   dev_err(&adapter->pdev->dev, "Unable to allocate MSI-X interrupts.\n");
   kfree(adapter->msix_entries);
   adapter->msix_entries = NULL;
-  err = -EIO;
- } else {
-  /* Adjust for only the vectors we'll use, which is minimum
-   * of max_msix_q_vectors + NONQ_VECS, or the number of
-   * vectors we were allocated.
-   */
-  adapter->num_msix_vectors = vectors;
+  return err;
  }
- return err;
+
+ /* Adjust for only the vectors we'll use, which is minimum
+  * of max_msix_q_vectors + NONQ_VECS, or the number of
+  * vectors we were allocated.
+  */
+ adapter->num_msix_vectors = err;
+ return 0;
 }
 
 /**
-- 
1.7.1